webui/src/app/charts/bar-chart/bar-chart.component.ts

```diff
@@ -2,21 +2,39 @@ import {
   Component,
   ElementRef,
   Input,
-  OnChanges,
   OnInit,
-  SimpleChanges
+  OnDestroy,
+  ChangeDetectorRef
 } from '@angular/core';
 import { axisBottom, axisLeft, max, scaleBand, scaleLinear, select } from 'd3';
 import { format } from 'd3-format';
 import * as _ from 'lodash';
 import { WindowService } from '../../services/window.service';
 
+interface DataModel {
+  code: string;
+  count: number;
+}
+
 @Component({
   selector: 'app-bar-chart',
   templateUrl: './bar-chart.component.html'
 })
-export class BarChartComponent implements OnInit, OnChanges {
-  @Input() value: any;
+export class BarChartComponent implements OnInit, OnDestroy {
+  @Input() set value(data: DataModel[] | null) {
```

Comment:
```suggestion
  @Input() set value(data?: DataModel[]) {
```

Reply:
This is case before the changing, however in setter case with `async` input, `null` is exact the first value been passed in.